addons/mail/static/src/web/discuss_app/sidebar.js

```diff
@@ -134,9 +134,11 @@ export class Sidebar extends Component {
         return category.threads.filter((threadLocalId) => {
             const thread = this.store.threads[threadLocalId];
             return (
-                (thread.is_pinned || thread.group_based_subscription) &&
+                (["channel", "group"].includes(thread.type)
+                    ? thread.hasSelfAsMember
+                    : thread.is_pinned) &&
```

Comment:
This part should be moved to a thread getter instead of duplicated.

Reply:
Note: `hasSelfAsMember` is not very reliable (if the member is not known, as we don't always fetch/send it)

Reply:
I would suggest for that case to do `hasSelfAsMember || is_pinned` maybe.

Reply:
then the condition will be just `thread.hasSelfAsMember || thread.is_pinned` if check `is_pinned` when channel type is `group` and `channel`

Reply:
And then i would just change the getter `hasSelfAsMember` and add `thread.is_pinned` into that since the original check in python is to check `_get_channels_as_member` as well.
And then the whole condition for sidebar and messaging menu will just be `thread.hasSelfAsMember`